# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [3]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [4]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3        V4        V5         V6        V7       
1 0.02084238 0.2922819 0.8066557 0.9130626 0.09344787 0.1019706 0.5281141
2 0.25636124 0.9045094 0.0305180 0.4435667 0.94777005 0.6376179 0.3858242
  V8        V9        V10       V11       V12        V13       V14      
1 0.3906494 0.4921655 0.1015893 0.2067715 0.62058720 0.2832025 0.3913543
2 0.7594683 0.2029220 0.7256194 0.3255878 0.07645985 0.4372985 0.3455614
  V15       V16       V17       V18       V19       V20      
1 0.4070162 0.5262919 0.4787672 0.9355555 0.2092072 0.7536532
2 0.8900791 0.5116049 0.3925067 0.7284477 0.9413976 0.3624088

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>%
    gather(Key,Value,V1:V3)
  

head(df2)

V4        V5         V6        V7        V8        V9        V10      
1 0.9130626 0.09344787 0.1019706 0.5281141 0.3906494 0.4921655 0.1015893
2 0.4435667 0.94777005 0.6376179 0.3858242 0.7594683 0.2029220 0.7256194
3 0.9130626 0.09344787 0.1019706 0.5281141 0.3906494 0.4921655 0.1015893
4 0.4435667 0.94777005 0.6376179 0.3858242 0.7594683 0.2029220 0.7256194
5 0.9130626 0.09344787 0.1019706 0.5281141 0.3906494 0.4921655 0.1015893
6 0.4435667 0.94777005 0.6376179 0.3858242 0.7594683 0.2029220 0.7256194
  V11       V12        V13       V14       V15       V16       V17      
1 0.2067715 0.62058720 0.2832025 0.3913543 0.4070162 0.5262919 0.4787672
2 0.3255878 0.07645985 0.4372985 0.3455614 0.8900791 0.5116049 0.3925067
3 0.2067715 0.62058720 0.2832025 0.3913543 0.4070162 0.5262919 0.4787672
4 0.3255878 0.07645985 0.4372985 0.3455614 0.8900791 0.5116049 0.3925067
5 0.2067715 0.62058720 0.2832025 0.3913543 0.4070162 0.5262919 0.4787672
6 0.3255878 0.07645985 0.4372985 0.3455614 0.8900791 0.5116049 0.3925067
  V18       V19       V20       Key Value     
1 0.9355555 0.2092072 0.7536532 V1  0.02084238
2 0.7284477 0.9413976 0.3624088 V1  0.25636124
3 0.9355555 0.2092072 0.7536532 V2  0.29228188
4 0.7284477 0.9413976 0.3624088 V2  0.90450935
5 0.9355555 0.2092072 0.7536532 V3  0.80665573
6 0.7284477 0.9413976 0.3624088 V3  0.03051800

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [7]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- merge(df3, df4, all="true")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook